src/Internal/Types/ScriptLookups.purs

```diff
@@ -803,6 +811,42 @@ addOwnOutput (OutputConstraint { datum: d, value }) = do
     ExceptT $ addDatum dat
     _valueSpentBalancesOutputs <>= provideValue value'
 
+resumeTimeConstraints
+  :: Array TxConstraint -> Either MkUnbalancedTxError (Array TxConstraint)
+resumeTimeConstraints constraints =
+  let
+    { no: nonTimeConstraints, yes: timeConstraints } = partition
+      isTimeConstraint
+      constraints
+    intervals = mapMaybe constraintToInterval timeConstraints
+  in
+    do
+      newInterval <- foldM mergeIntervals always intervals
+      pure $ cons (MustValidateIn newInterval) nonTimeConstraints
```

Comment:
It's a shame we go over the constraints twice, [`partitionMap`](https://pursuit.purescript.org/packages/purescript-filterable/5.0.0/docs/Data.Filterable#v:partitionMap) gives us what we need but it's not worth pulling in just for this

```suggestion
resumeTimeConstraints constraints = do
  let
    { no: nonTimeConstraints, yes: timeConstraints } = partition
      isTimeConstraint
      constraints
    intervals = mapMaybe constraintToInterval timeConstraints
  newInterval <- foldM mergeIntervals always intervals
  pure $ cons (MustValidateIn newInterval) nonTimeConstraints
```